cirq/contrib/two_qubit_gates/gate_compilation_test.py

```diff
@@ -0,0 +1,95 @@
+"""Tests for gate_compilation.py"""
+import numpy as np
+import pytest
+
+from cirq import unitary, FSimGate, value
+from cirq.contrib.two_qubit_gates import example
+from cirq.contrib.two_qubit_gates.gate_compilation import (
+    gate_product_tabulation)
+from cirq.contrib.two_qubit_gates.math_utils import (
+    random_two_qubit_unitaries_and_kak_vecs, unitary_entanglement_fidelity,
+    weyl_chamber_mesh, random_qubit_unitary)
+
+_rng = value.parse_random_state(11)  # for determinism
+
+sycamore_tabulation = gate_product_tabulation(unitary(
+    FSimGate(np.pi / 2, np.pi / 6)),
+                                              0.2,
+                                              random_state=_rng)
+
+sqrt_iswap_tabulation = gate_product_tabulation(unitary(
+    FSimGate(np.pi / 4, np.pi / 24)),
+                                                0.1,
+                                                random_state=_rng)
+
+_random_2Q_unitaries, _ = random_two_qubit_unitaries_and_kak_vecs(100, _rng)
+
+
+@pytest.mark.parametrize('tabulation',
+                         [sycamore_tabulation, sqrt_iswap_tabulation])
+@pytest.mark.parametrize('target', _random_2Q_unitaries)
+def test_gate_compilation_matches_expected_max_infidelity(tabulation, target):
+    result = tabulation.compile_two_qubit_gate(target)
+
+    if result.success:
```

Comment:
Good point. For the very loose fidelity bound it looks like it's always successful. In other cases (i.e., a poor chose of base gate) it may only be successful some of the time.